core/types/rollup_cost.go

```diff
@@ -259,11 +295,23 @@ func intToScaledFloat(scalar *big.Int) *big.Float {
 
 // extractL1GasParams extracts the gas parameters necessary to compute gas costs from L1 block info
 func extractL1GasParams(config *params.ChainConfig, time uint64, data []byte) (gasParams, error) {
-	// edge case: for the very first Ecotone block we still need to use the Bedrock
-	// function. We detect this edge case by seeing if the function selector is the old one
-	// If so, fall through to the pre-ecotone format
-	// Both Ecotone and Fjord use the same function selector
-	if config.IsEcotone(time) && len(data) >= 4 && !bytes.Equal(data[0:4], BedrockL1AttributesSelector) {
+	if config.IsIsthmus(time) {
```

Comment:
We will need the same sort of hack we have in here for is ecotone. Basically for the first block, it wont be the calldata of isthmus style since the contract deployment happens then the next block it sends the upgrade tx with the new style.

Reply:
Done